includes/core/server/class-request.php

```diff
@@ -0,0 +1,523 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server;
+
+use WC_Payments;
+use WC_Payments_Http_Interface;
+use WC_Payments_API_Client;
+use WCPay\Core\Exceptions\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Immutable_Parameter_Exception;
+use WCPay\Core\Exceptions\Invalid_Request_Parameter_Exception;
+
+/**
+ * Base for requests to the WCPay server.
+ */
+abstract class Request {
+	/**
+	 * Contains a set of params, which the class considers immutable by others.
+	 *
+	 * Overwrite this in your class for individual properties.
```

Comment:
Small nitpick but can we add the same description lines to the 3 constants below?